optuna/study/_multi_objective.py

```diff
@@ -91,26 +91,31 @@ def _fast_non_dominated_sort(
                 len(penalty), len(objective_values)
             )
         )
-    nondomination_rank = np.zeros(len(objective_values), dtype=int)
+    nondomination_rank = np.full(len(objective_values), -1)
     is_nan = np.isnan(penalty)
+    n_below = n_below or len(objective_values)
 
     # First, we calculate the domination rank for feasible trials.
     is_feasible = np.logical_and(~is_nan, penalty <= 0)
-    ranks, feasible_bottom_rank = _calculate_nondomination_rank(
+    ranks, bottom_rank = _calculate_nondomination_rank(
         objective_values[is_feasible], n_below=n_below
     )
-    nondomination_rank[is_feasible] += ranks
+    nondomination_rank[is_feasible] += 1 + ranks
```

Comment:
Do we start the rank from 1 now?
Probably, it should be documented somewhere.

Reply:
I added a description of current implementation to the docstrings.